lib/iris/cube.py

```diff
@@ -3078,26 +3040,12 @@ def __copy__(self):
     def __deepcopy__(self, memo):
         return self._deepcopy(memo)
 
-    def _deepcopy(self, memo, data=None):
-        if data is None:
-            # Use a copy of the source cube data.
-            if self.has_lazy_data():
-                # Use copy.copy, as lazy arrays don't have a copy method.
-                new_cube_data = copy.copy(self.lazy_data())
-            else:
-                # Do *not* use copy.copy, as NumPy 0-d arrays do that wrong.
-                new_cube_data = self.data.copy()
-        else:
-            # Use the provided data (without copying it).
-            if not is_lazy_data(data):
-                data = np.asanyarray(data)
-
-            if data.shape != self.shape:
-                msg = 'Cannot copy cube with new data of a different shape ' \
-                      '(slice or subset the cube first).'
-                raise ValueError(msg)
+    def _deepcopy(self, memo, data=None, dtype='none', fill_value='none'):
+        dm = self._data_manager.copy(data=data, realised_dtype=dtype)
 
-            new_cube_data = data
+        if isinstance(fill_value, six.string_types) and \
+                fill_value == 'none':
```

Comment:
Aha! This seemed like an odd over-specification of the logical test, so it is good to rediscover why it was necessary (and hurrah for tests doing their job!). Perhaps a little comment about this would be good so that someone doesn't come along in the future in a well-meaning manner and change it again...